transect-line.py

```diff
@@ -0,0 +1,99 @@
+import threading
+import time
+
+from mcu_lib.mcu import MCUInterface
+from mcu_lib.command_constants import *
+from comms import Communications
+from controls_pyqt.key_signal import KeySignal
+
+
+class TransectLine:
+	def __init__(self, comms: Communications, mcu_object: MCUInterface):
+    	self.comms: Communications = comms
+    	self.mcu: MCUInterface = mcu_object
+	self.HEIGHT_THRESH: int = 1				# CHANGE THIS
+	self.ORIENTATION_THRESH: int = 5		# CHANGE THIS
+	self.DISTANCE: int = 6					# CHANGE THIS
+	self.running = True
+			
+	def move_forward(self):
+		trigger: str = "w"
+		pressed: bool = True
+		starttime = time.time()
+		self.comms.read_send(KeySignal(trigger, pressed))
+		self.distance_travelled: float = 0
+		
+		while(self.distance_travelled < self.DISTANCE):
+			timenow = time.time()
+			time_diff: float = timenow - starttime
+			starttime = timenow
+			
+			FORWARD_AXIS_INDEX: int = 0		# CHANGE THIS
+			accel: float = self.mcu.latest_accel[FORWARD_AXIS_INDEX]
+			
+			self.distance_travelled = self.distance_travelled + 0.5 * accel * (time_diff ** 2)
+		
+		pressed = False
+		self.comms.read_send(KeySignal(trigger, pressed))
+		self.running = False
+		
+	def move(self, time_needed: float, trigger: str):
+		pressed: bool = True
+		self.comms.read_send(KeySignal(trigger, pressed))
```

Comment:
suggestion: `read_send` is dumb when you can use `up()`, `down()`, `tilt_up()`, `turn_left()`, etc.